wp-content/themes/core/components/link/Link_Controller.php

```diff
@@ -14,6 +14,8 @@ class Link_Controller extends Abstract_Controller {
 	public const ARIA_LABEL     = 'aria_label';
 	public const CLASSES        = 'classes';
 	public const ATTRS          = 'attrs';
+	public const ICON_CLASSES   = 'icon_classes';
+	public const IS_ICON_BEFORE = 'is_icon_before';
```

Comment:
Going to replace with a :before :after psudo classes

Reply:
NOW I REMEMBER WHY I DID IT THIS WAY. The `.icon` class applies a custom font, which means I can't just simply apply it to the A tag because it messes with its font and I would need to override it. So instead I applied it to  sub element like a span rather than the A tag itself.

Any idea how to overcome this shortcoming?

```
<a><i></i><a/>
```

Reply:
You can add the icon mixin to the before/after pseudo element. Here's [an example from Durham](https://github.com/moderntribe/durham-communities/blob/1a1ddbcde8c31ff4c99574b172f05f6de02c6ac2/wp-content/themes/core/components/navigation/css/navigation.pcss#L100-L109).